src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -2044,7 +2072,22 @@ private async ValueTask<JsonRpcMessage> DispatchIncomingRequestAsync(JsonRpcRequ
                     }
                 }
 
-                return await this.DispatchRequestAsync(request, targetMethod, cancellationToken).ConfigureAwait(false);
+                if (this.JoinableTaskFactory is null || request.TryGetTopLevelProperty<string>(JoinableTaskTokenHeaderName, out string? parentToken) is false || parentToken is null)
+                {
+                    return await this.DispatchRequestAsync(request, targetMethod, cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    JsonRpcMessage? result = null;
+                    await this.JoinableTaskFactory.RunAsync(
```

Comment:
This closure pattern suggests adding a `RunAsync<T>` overload would probably be a worthy improvement.